### PR TITLE
[C10D][Logger]Add more info to c10d logger

### PR DESCRIPTION
**Pull request by wz337**
This PR adds pg_name and world_size to c10d logging.

### CODE REVIEW

**Review by kumpera — APPROVED** [review]
Looks good, thanks for the improvements!

File: torch/distributed/c10d_logger.py [hunk 1]
```diff
@@ -40,30 +41,37 @@ def _get_logging_handler(destination: str = "default") -> Tuple[logging.Handler,
 _c10d_logger = _get_or_create_logger()
 
 
+def _get_msg_dict(func_name, *args, **kwargs) -> Dict[str, Any]:
+    if dist.is_initialized():
+        msg_dict = {
+            "func_name": f"{func_name}",
+            "args": f"{args}, {kwargs}",
+            "pg_name": f"{dist._get_process_group_name(kwargs.get('pg'))}",  # type: ignore[arg-type]
```
Comment (fduwjj) [inline torch/distributed/c10d_logger.py:49]: what will the pg_name look like?

Comment (wz337) [reply torch/distributed/c10d_logger.py:49]: pg_name is derived from here: https://fburl.com/7rqfahug

**Pull request closed without merging**